assignment.md

```diff
@@ -11,47 +11,75 @@ If the version control concepts or `git` commands are still confusing to you, co
  
     ```bash
-    # list the new list of remotes here
+   git remote -v                                                      
+   origin	https://github.com/emilyolafson/version_control_assignment (fetch)
+   origin	https://github.com/emilyolafson/version_control_assignment (push)
+   upstream	https://github.com/WCM-datascibasics/version_control_assignment (fetch)
+   upstream	https://github.com/WCM-datascibasics/version_control_assignment (push)  
     ```
 
  3. Save your modifications to this file so far and create a commit indicating you've answered the first 2 questions. Then try pushing the changes to the `upstream` destination on GitHub. What happens? Explain in your own words why does this happen? What are the benefits of having this `upstream` remote added when working collaboratively?
 
-    > Your answer here
+    > control who can change things
 
 
  4. Now clone the repository for the [class website](https://github.com/WCM-datascibasics/wcm-datascibasics.github.io), and in the class website repository:
     - a. Explore the version history by visualizing it as a graph.
         ```bash
-        # git command and output here
-        ```
+         git log --graph --decorate --oneline      
+           ```
 
     - b. Who was the last person to modify `README.md`? (Hint: use `git log` with an argument)
         ```bash
-        # git command and output here
+        git log -p README.md
+
+         commit 00e8ae51a08a0f3754ab228f62b51651595c4426
+         Author: Xihe Xie <axiezai@gmail.com>
+         Date:   Mon Dec 28 16:01:51 2020 -0500
+
+            readme and try new size
+
+         diff --git a/README.md b/README.md
+         new file mode 100644
+         index 0000000..dbad2e5
+         --- /dev/null
+         +++ b/README.md
+         @@ -0,0 +1 @@
+         +# Course GitHub Pages repo
+
         ```
     
-    - c. What was the commit message associated with the last modification to the `-Week 2` line of `index.md` file? (Hint: use `git blame` and `git show`, check out their help pages if you are not sure what to do)
+    - c. What was the commit messa ge associated with the last modification to the `-Week 2` line of `index.md` file? (Hint: use `git blame` and `git show`, check out their help pages if you are not sure what to do)
+        added ch2 assignment
 
  5. In the course website repository (or another repository you want to clone from GitHub), modify one of its existing files:
      - a. What happens when you do `git stash`?
-       > Answer here
+       > Saved working directory and index state WIP on master
      - b. What do you see when you run `git log --all --oneline` after stashing your modification?
-       > Answer here
+       > 7582237 (refs/stash) WIP on master: f83956f added week 8
      - c. Run `git stash pop` to undo what you did with `stash`, in what scenario might this be useful?
-       > Answer here
+       > make a mistake
```

Comment:
`git stash pop` actually merges the stashed changes, so no, not for mistakes... you won't merge a mistake back to `HEAD`. Sure `stash`ing an error is useful, but I'm asking specifically for `pop`, and your answer is simply not appropriate in this scenario. -2%. 

`git stash drop` drops your stashed changes.